src/makeInfinityAdder.js

```diff
@@ -1,10 +1,25 @@
 'use strict';
 
-/***
+/*
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(n) {
+    const totalSum = sum;
+
+    if (n === undefined) {
```

Comment:
The check for `n === undefined` is used to determine when to reset the sum. This is a valid approach, but consider using a more explicit method to reset, such as a specific method call, to avoid potential issues with falsy values.